dev-docs/development_process.md

```diff
@@ -76,63 +76,141 @@ export DOCKER_BUILDKIT=1
-call “dev deploy”. The command to invoke this is
+call a “dev deploy”. The command to invoke a dev deploy is
 
-```
-hailctl dev deploy -b <github_user_name>/hail:<branch_name> -s <step1>,<step2>,...
+```bash
+HAIL_DEFAULT_NAMESPACE=default hailctl dev deploy -b <github_user_name>/hail:<branch_name> -s <step1>,<step2>,...
 ```
 
 Dev deploy creates a batch that deploys the build steps specified by the `-s` in
-your own Kubernetes namespace, MySQL database. For example, if we want to test
+your Kubernetes dev namespace. For example, if we want to test
 whether the Batch tests still pass, we would specify -s test_batch. This will
 run all the dependent steps for testing Batch such as creating credentials,
 a live Auth service, a MySQL database for Batch, and a live Batch deployment.
-Your namespace name is the same as your username.
+
 Submitting a dev deploy with hailctl will give you the link to a UI
 where you can monitor the progress of everything deploying and get the logs for
 any steps that fail. You can also see a recent history of your dev deploys at
 [ci.hail.is/me](https://ci.hail.is/me).
 
+The first time a namespace is used, the Auth service in that namespace won't
+have any users. In order submit jobs to the namespace, make sure to include
+the `add_developers` step to the dev deploy. So a first dev deploy might look like:
 
-If the tests fail, you can then examine the Kubernetes logs for the service
-using something like
+```bash
+HAIL_DEFAULT_NAMESPACE=default hailctl dev deploy -b <github_user_name>/hail:<branch_name> -s deploy_batch,add_developers
+```
 
+After this dev deploy completes, you should be able to access your namespace
+by navigating to https://internal.hail.is/<username>/batch.
+
+To submit jobs to your dev namespace, you need to configure your local hail
+installation to point to the dev namespace. You can do this by running
+
+```bash
+hailctl dev config set default_namespace <my_namespace>
+hailctl auth login
 ```
+Then, any Hail Batch or Query-on-Batch script will submit to your dev namespace.
+
+You can also use the `HAIL_DEFAULT_NAMESPACE` environment variable to alter
+the namespace for individual terminal sessions or commands.
+
+NOTE: The dev deploy command above sets `HAIL_DEFAULT_NAMESPACE=default`. That's
+because `hailctl dev deploy` submits a request to the production CI service that
+lives in the `default` namespace. This does *not* mean that your branch will be deployed
+to the `default` namespace, it will always be deployed into your dev namespace.
+
+To examine the Kubernetes logs for the service in your namespace, use something like
+
+```bash
 kubectl -n <my_namespace> logs -l app=batch-driver --tail=999999 | less
 ```
 
-To check the MySQL database, you first need to find the name of the specific pod
-running an “admin-pod” and then log into that pod:
+To check the MySQL database in your namespace, you can log in to the database
```

Comment:
I think either here or somewhere else or as a todo item, we should talk about how to delete a test database like `username-batch` so it will be recreated when testing migrations. I know this was a known wrinkle.

Reply:
Great point, maybe I'll follow up with a development FAQ? I don't want to make this getting started section too long but I feel like there's multiple points like this one that would be nice to have in one place.

Reply:
sounds good.